server/src/main/java/org/elasticsearch/action/admin/indices/template/get/GetIndexTemplatesRequest.java

```diff
@@ -9,38 +9,40 @@
 package org.elasticsearch.action.admin.indices.template.get;
 
 import org.elasticsearch.action.ActionRequestValidationException;
-import org.elasticsearch.action.support.master.MasterNodeReadRequest;
+import org.elasticsearch.action.support.local.LocalClusterStateRequest;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
 
 import java.io.IOException;
+import java.util.Map;
 
 import static org.elasticsearch.action.ValidateActions.addValidationError;
 
 /**
  * Request that allows to retrieve index templates
  */
-public class GetIndexTemplatesRequest extends MasterNodeReadRequest<GetIndexTemplatesRequest> {
+public class GetIndexTemplatesRequest extends LocalClusterStateRequest {
 
     private final String[] names;
 
-    public GetIndexTemplatesRequest(String... names) {
-        super(TRAPPY_IMPLICIT_DEFAULT_MASTER_NODE_TIMEOUT);
+    public GetIndexTemplatesRequest(TimeValue masterTimeout, String... names) {
+        super(masterTimeout);
         this.names = names;
     }
 
+    /**
+     * NB prior to 9.0 this was a TransportMasterNodeReadAction so for BwC we must remain able to read these requests until
+     * we no longer need to support calling this action remotely.
+     */
```

Comment:
Normally you've been adding an `@UpdateForV10(…)`, should we add one here also?

Reply:
Good catch, thanks!